flink-ml-lib/src/main/java/org/apache/flink/ml/anomalydetection/isolationforest/IForest.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.anomalydetection.isolationforest;
+
+import org.apache.flink.ml.linalg.DenseVector;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+/** Construct isolation forest. */
+public class IForest implements Serializable {
+    public final int numTrees;
+    public List<ITree> iTreeList;
+    public Double center0;
+    public Double center1;
+    public int subSamplesSize;
+
+    public IForest(int numTrees) {
+        this.numTrees = numTrees;
+        this.iTreeList = new ArrayList<>(256);
+        this.center0 = null;
+        this.center1 = null;
+    }
+
+    public void generateIsolationForest(DenseVector[] samplesData, int[] featureIndices) {
+        int n = samplesData.length;
+        this.subSamplesSize = Math.min(256, n);
```

Comment:
nits: would it be more consistent with existing code to use `subSamplesSize` directly here?

Same for other usages of `this.*` outside constructor.

Reply:
OK, I will fix it.